### PR TITLE
Hooked up output-format flag to cargo-kani script

### LINKED ISSUE #814: The `--output-format` option does not work with cargo kani
Steps to reproduce:
1. cargo new foo
2. cd foo
3. cargo kani --output-format regular

with Kani version: 1a414349a41

I expected to see the new output format.

Instead, this happened: The output was dumped using the old format:
```
** Results:
[assertion.1] Code generation sanity check: Correct CBMC vtable size for StructTag("tag-_4317141983261250885") (MIR type Closure(DefId(2:9147 ~ std[2f3c]::rt::lang_start::{closure#0}), [(), i8, extern "rust-call" fn(()) -> i32, (fn(),)])). Please report failures:
https://github.com/model-checking/kani/issues/new?template=bug_report.md: SUCCESS
[pointer_primitives.1] dead object in OBJECT_SIZE(&temp_0): SUCCESS
function _ZN3std2io5stdio6_print17hbd655a505d65b63eE
[_ZN3std2io5stdio6_print17hbd655a505d65b63eE.assertion.1] assertion false: FAILURE

/rustc/9ad5d82f822b3cb67637f11be2e65c5662b66ec0/library/core/src/fmt/mod.rs function std::fmt::Arguments::<'a>::new_v1
[std::fmt::Arguments::<'a>::new_v1.assertion.1] line 363 This is a placeholder message; Kani doesn't support message formatted at runtime: SUCCESS
[std::fmt::Arguments::<'a>::new_v1.pointer_arithmetic.1] line 365 pointer arithmetic: dead object in (unsigned char *)&var_24 + 0: SUCCESS
[std::fmt::Arguments::<'a>::new_v1.pointer_dereference.1] line 365 dereference failure: dead object in *((unsigned char **)((unsigned char *)&var_24 + 0)): SUCCESS

** 1 of 6 failed (2 iterations)
VERIFICATION FAILED
```

### PR DESCRIPTION
**Pull request by zhassan-aws**
### Description of changes: 

The `--output-format` flag does not work with `cargo kani`. This PR fixes this issue.

### Resolved issues:

Resolves #814 

### Testing:

* How is this change tested? Added a new test

* Is this a refactor change? No

### Checklist
- [X] Each commit message has a non-empty body, explaining why the change was made
- [X] Methods or procedures are documented
- [X] Regression or unit tests are included, or existing tests cover the modified code
- [X] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: tests/cargo-kani/output-format/main.expected [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Description: "assertion failed: 1 + 1 == 2"
+Description: "dead object in OBJECT_SIZE(&temp_0)"
```
Comment (jaisnan) [inline tests/cargo-kani/output-format/main.expected:2]: Are we making the new output the default output for Cargo Kani?

Comment (zhassan-aws) [reply tests/cargo-kani/output-format/main.expected:2]: No, this test uses `--output-format regular`:
```
[package.metadata.kani]
flags = {output-format = "regular"}
```

File: library/kani_macros/src/lib.rs [hunk 2]
```diff
@@ -13,7 +13,7 @@ use proc_macro::TokenStream;
 
 #[cfg(all(not(kani), not(test)))]
 #[proc_macro_attribute]
-pub fn proof(_attr: TokenStream, item: TokenStream) -> TokenStream {
+pub fn proof(_attr: TokenStream, _item: TokenStream) -> TokenStream {
```
Comment (celinval) [inline library/kani_macros/src/lib.rs:16]: :)

File: tests/cargo-kani/output-format/main.expected (hunk 1 above)
Comment (jaisnan) [reply tests/cargo-kani/output-format/main.expected:2]: Thanks, maybe it's time to make it the default output if so much new work depends on it.

**Review by celinval — APPROVED** [review]

**Pull request merged**